task_manager/task_manager.hh

```diff
@@ -0,0 +1,323 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include "replica/database.hh"
+
+using task_id = utils::tagged_uuid<struct task_id_tag>;
+using is_abortable = bool_class <struct abortable_tag>;
+
+class task_manager : public seastar::peering_sharded_service<task_manager> {
+public:
+    class task;
+    class module;
+    struct config {
+        utils::updateable_value<uint32_t> task_ttl = utils::updateable_value<uint32_t>(0);
+    };
+    using task_ptr = seastar::shared_ptr<task_manager::task>;
```

Comment:
Any reason not to use `lw_shard_ptr`?

Reply:
I think that currently since many classes can inherit from `task_manager::task` we cannot use `lw_shared_ptr` here. Problem disappears with `task_manager::task::impl` though.